appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/servlet/HealthCheckServlet.java

```diff
@@ -77,8 +83,28 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
             return;
         }
 
-        healthCheckService.performHealthChecks(response, HealthCheckType.fromPath(request.getPathInfo()));
+        for (Checker checker : payaraHealthCheckService.getConfiguration().getCheckerList()) {
```

Comment:
I think this whole loop does not belong in this class. If a `Checker` is configured to also occur in MP HealthCheck it should add a `HealthCheck` implementation to the service which is then processed like all other registered checks. If the  `Checker` configuration changes and the Payara health check no longer is exposed in MP the health check should be de-registered as part of the reaction to the configuration change, for example though listening to configuration change events.